add ability to split config into 2 files

MaxFedotov opened the pull request:
Added ability to create two different config files, if variable `$split_config`is set to true - one for admin\mysql variables (controlled by `$config_file` and `$manage_config_file` variables), another for servers\users\hostgroups\scheduler\rules params of ProxySQL (controlled by `$proxy_config_file` and `manage_proxy_config_file` variables) .  This was done to allow management of main config file by puppet and manage all proxy configuration settings by some external provider (for example, consul-template, bash script), as this file needs to be updated in more frequent intervals that actual puppet agent interval (e.g. for each failover)
Also added creation of `sys_owner` user and `sys_group` group in install.pp to fix possible errors if this groups doesn't exists.
Added `package_name` and `package_ensure`, because got problems with mariadb conflicting package trying to install during puppet run.
Added `reload_config` manifest to allow dynamic config reload

MaxFedotov commented on manifests/reload_config.pp:
```diff
@@ -0,0 +1,26 @@
+#
+# This class is called from proxysql to update config if it changed.
+#
+class proxysql::reload_config {
```
it will also require to stop proxysql first, which is not a good way, as many variables can be just change in a runtime. 

All configurations related to users/servers/... will be reloaded by corresponding provider on change, so i think this functionality don't need to be duplicated here

mcrauwel replied:
yeah I know, but then I wonder why use the configuration file to make updates to the configuration? just use the internal database to configure the proxy?

and when you say the configurations to users/server/... will be reloaded by corresponding provider on change, there is the `proxy_global_variable` to do the same for variables ¯\_(ツ)_/¯

MaxFedotov replied:
Sure, there is proxy_global_variable and you can configure it all on the fly. But some times it is a good idea to have such static variables as config stored in a file - it simplifies management

MaxFedotov commented on manifests/install.pp:
```diff
@@ -27,6 +36,8 @@
   }
 
   class { '::mysql::client':
+    package_name    => 'mysql-client',
```
Answered in https://github.com/voxpupuli/puppet-proxysql/pull/58

mcrauwel replied:
I would remove it from this PR and keep the fix in #58. Agreed?

MaxFedotov replied:
Removed

bastelfreak commented on manifests/init.pp:
```diff
@@ -132,8 +143,11 @@
   String $monitor_username = $::proxysql::params::monitor_username,
   Sensitive[String] $monitor_password = $::proxysql::params::monitor_password,
 
+  Boolean $split_config = $::proxysql::params::split_config,
```
please remove the leading `::`

MaxFedotov commented:
Review notes applied - changed all $:: to $. Changed to facts[] instead of $::. Changed scope.lookupvar() in proxysql.cnf.erb. Moved proxysql_proxy.cnf.erb to proxysql_proxy.cnf.epp

mcrauwel commented:
@MaxFedotov can you rebase on the current master?

mcrauwel commented:
now let's rebase this. I will try to find the issue in the tests...

MaxFedotov commented:
done, but errors are the same

bastelfreak commented:
![2018-09-03-175609_602x349_scrot](https://user-images.githubusercontent.com/341866/44995275-a86bcd00-afa2-11e8-9fb2-1fb832e9aab8.png)

mhm, travis is trolling me at the moment.

bastelfreak commented on manifests/install.pp:
```diff
@@ -35,6 +35,15 @@
+    ensure => 'present',
+  }
+
+  user { $::proxysql::sys_owner:
```
please do not add leading `::`

bastelfreak commented on manifests/init.pp:
```diff
@@ -158,6 +168,11 @@
 
+  Boolean $split_config = $proxysql::params::split_config,
+
+  String $proxy_config_file = $proxysql::params::proxy_config_file,
```
Can you please use `String[1]`? Otherwise empty string is allowed as well.

bastelfreak replied:
Oh this is a path. Please use Stdlib::Absolutepath here.

bastelfreak commented on manifests/reload_config.pp:
```diff
@@ -0,0 +1,26 @@
+          SAVE MYSQL VARIABLES TO DISK; \"
+        ",
+        subscribe   => $subscribe,
+        require     => [ Service[$::proxysql::service_name] , File['root-mycnf-file'] ],
```
Please do not add leading `::`

MaxFedotov commented:
Still got this :(
  Failure/Error: it { is_expected.to compile.with_all_deps }
       dependency cycles found: (File[/etc/apt/sources.list.d/proxysql_repo.list] => Apt::Setting[list-proxysql_repo] => Apt::Source[proxysql_repo] => Class[Proxysql::Repo] => Class[Proxysql::Install] => Group[root] => File[/etc/apt/sources.list.d/proxysql_repo.list])

MaxFedotov commented:
@bastelfreak were you able to fix problems with Travis?

bastelfreak commented on metadata.json:
```diff
@@ -12,6 +12,10 @@
     },
+    {
+      "name": "puppetlabs-stdlib",
+      "version_requirement": ">= 1.0.0"
```
We need to remove this now or wait for releases from puppetlabs: https://github.com/voxpupuli/puppet-proxysql/issues/65

MaxFedotov replied:
Let's remove this for now and i will create another PR, which will add this and can be merged later :)

bastelfreak commented:
@mcrauwel can you look into this? Is master broken?

mcrauwel commented:
master is ok, so it must be something in the PR, it's only happening on deb based systems hence some tests (RHEL/CentOS) are successful

MaxFedotov commented:
I have no errors when installing it on centos. Will check PR one more time again tomorrow

mcrauwel commented:
current issue looks like a bug in the debian8 package for proxysql...

mcrauwel commented:
Let’s wait for a day or 2 to see if they can quickly fix it?

Sent from my iPhone

> On 8 Sep 2018, at 20:23, Tim Meusel <notifications@github.com> wrote:
> 
> should we, for now, remove debian 8 from .travis.yml/metadata.json? So we could merge this now and later add it back. This would allow us to do a new release finally.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

mcrauwel commented:
update: i found the error and opened an upstream PR sysown/proxysql#1680

mcrauwel commented:
Hooray! Thanks for your contributions @MaxFedotov and thanks for the help @bastelfreak!
